task_processing/plugins/mesos/execution_framework.py

```diff
@@ -361,7 +361,10 @@ def statusUpdate(self, driver, update):
             task=task_id
         ))
 
-        self.task_update_queue.put(self.translator(update, task_id))
+        md = self.task_metadata[task_id]
```

Comment:
This is a really nasty abstraction leak. I think we need a better way to communicate events and status updates up and down the executor stack. I like the idea of a queue of Events, but it probably needs to be multi-consumer single producer or some such. In java they have Event Listeners and notifiers, maybe we want something similar?